src/metabase/mbql/schema.clj

```diff
@@ -584,7 +584,7 @@
     ;; with appropriate aliases.
     (s/optional-key :fields)       (s/cond-pre
                                     (s/enum :all :none)
-                                    (su/distinct (su/non-empty [joined-field])))
+                                    (su/distinct (su/non-empty [(s/either joined-field datetime-field)])))
```

Comment:
I know `s/either` is deprecated, but doing this with `s/conditional` is horrendous (I can see argument for writing our own `either-clause` specifically for  `defclause` that does the right thing and does `s/conditional` underneath.